cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterPumpDisassociateController.java

```diff
@@ -96,49 +86,38 @@ public WaterPumpDeleteController(MetricRegistry metrics) {
     public void handle(@NotNull Context ctx) {
         try (Timer.Context ignored = markAndTime(DELETE)) {
             DSLContext dsl = getDslContext(ctx);
-            boolean deleteAccounting = Boolean.parseBoolean(ctx.queryParam(DELETE));
+            boolean deleteAccounting = Boolean.parseBoolean(ctx.queryParam(METHOD));
             String officeId = ctx.pathParam(OFFICE);
             String projectName = ctx.pathParam(PROJECT_ID);
             String entityName = ctx.pathParam(WATER_USER);
             PumpType pumpType = PumpType.valueOf(ctx.queryParam(PUMP_TYPE));
             String contractName = ctx.pathParam(CONTRACT_NAME);
             WaterContractDao contractDao = getContractDao(dsl);
-            CwmsId projectLocation = buildCwmsId(officeId, projectName);
-            List<WaterUserContract> contract = contractDao.getAllWaterContracts(projectLocation, entityName);
-
-            for (WaterUserContract waterUserContract : contract) {
-                if (waterUserContract.getContractId().getName().equals(contractName)) {
-                    switch (pumpType) {
-                        case IN:
-                            contractDao.removePumpFromContract(waterUserContract,
-                                    waterUserContract.getPumpInLocation().getPumpLocation().getName(),
-                                    PumpType.IN, deleteAccounting);
-                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
-                            return;
-                        case OUT:
-                            contractDao.removePumpFromContract(waterUserContract,
-                                    waterUserContract.getPumpOutLocation().getPumpLocation().getName(),
-                                    PumpType.OUT, deleteAccounting);
-                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
-                            return;
-                        case BELOW:
-                            contractDao.removePumpFromContract(waterUserContract,
-                                    waterUserContract.getPumpOutBelowLocation().getPumpLocation().getName(),
-                                    PumpType.BELOW, deleteAccounting);
-                            ctx.status(HttpServletResponse.SC_NO_CONTENT);
-                            return;
-                        default:
-                            CdaError error = new CdaError("Invalid pump type provided.");
-                            LOGGER.log(Level.SEVERE, "Invalid pump type provided.");
-                            ctx.status(HttpServletResponse.SC_NOT_FOUND).json(error);
-                            return;
-                    }
+            CwmsId projectLocation = CwmsId.buildCwmsId(officeId, projectName);
+            WaterUserContract contract = contractDao.getWaterContract(contractName, projectLocation, entityName);
 
-                }
+            switch (pumpType) {
```

Comment:
parsing of the enum will ensure that you don't have an unsupported type

Reply:
removed switch statement